app/services/reports/facility_progress_service.rb

```diff
@@ -122,51 +131,87 @@ def dimension_combinations_for_v1(indicator)
       end
     end
 
+    memoize def monthly_total_follow_ups
+      @monthly_facility_data.each_with_object({}) do |record, hsh|
+        hsh[record.period] = if region.diabetes_management_enabled?
+          record[:monthly_follow_ups_htn_or_dm]
+        else
+          record[:monthly_follow_ups_htn_only] + record[:monthly_follow_ups_htn_and_dm]
+        end
+      end
+    end
+
+    memoize def monthly_total_registrations
+      @monthly_facility_data.each_with_object({}) do |record, hsh|
+        hsh[record.period] = if diabetes_enabled
+          record[:monthly_registrations_htn_or_dm]
+        else
+          record[:monthly_registrations_htn_only] + record[:monthly_registrations_htn_and_dm]
+        end
+      end
+    end
+
     memoize def daily_registrations_breakdown
-      @daily_facility_data.each_with_object({}) do |record, hsh|
+      registrations_breakdown(period_type: "daily", facility_data: @daily_facility_data)
+    end
+
+    memoize def daily_follow_ups_breakdown
+      follow_ups_breakdown(period_type: "daily", facility_data: @daily_facility_data)
+    end
+
+    memoize def monthly_registrations_breakdown
+      registrations_breakdown(period_type: "monthly", facility_data: @monthly_facility_data)
+    end
+
+    memoize def monthly_follow_ups_breakdown
+      follow_ups_breakdown(period_type: "monthly", facility_data: @monthly_facility_data)
+    end
+
+    memoize def registrations_breakdown(period_type:, facility_data:)
```

Comment:
The method's signature fits very well with the `daily_` and `monthly_` column names in the matviews. V nice 👌🏼

Reply:
Let's make these two methods private, since the other methods will act as the public interface.